manifest/v1alpha/direct/validation.go

```diff
@@ -383,6 +385,29 @@ var queryDelayGreaterThanOrEqualToDefaultValidationRule = govy.NewRule(func(spec
 	return nil
 }).WithErrorCode(errCodeQueryDelayGreaterThanOrEqualToDefault)
 
+const errorCodeUnsupportedReleaseChannel = "unsupported_release_channel"
+
+var releaseChannelValidationRule = govy.NewRule(func(spec Spec) error {
+	typ, _ := spec.GetType()
+	if typ != v1alpha.SplunkObservability && spec.ReleaseChannel == v1alpha.ReleaseChannelAlpha {
+		return govy.NewPropertyError(
+			"releaseChannel",
+			spec.ReleaseChannel,
+			errors.New("must be one of [stable, beta]"),
+		)
+	}
+
+	if typ == v1alpha.SplunkObservability && spec.ReleaseChannel != v1alpha.ReleaseChannelAlpha {
+		return govy.NewPropertyError(
+			"releaseChannel",
+			spec.ReleaseChannel,
+			errors.New("must be alpha for Splunk Observability"),
```

Comment:
just to highlight the expected value better:

```suggestion
			errors.New("must be 'alpha' for Splunk Observability"),
```